src/main/java/net/fabricmc/loom/configuration/providers/mappings/IntermediateMappingsService.java

```diff
@@ -80,7 +82,11 @@ public static IntermediateMappingsService create(IntermediateMappingsProvider in
 			throw new UncheckedIOException("Failed to provide intermediate mappings", e);
 		}
 
-		return new IntermediateMappingsService(intermediaryTiny);
+		String expectedSrcNs = minecraftProvider.canMergeJars()
+				? MappingsNamespace.OFFICIAL.toString() // >=1.3
+				: MappingsNamespace.INTERMEDIARY.toString(); // <1.3
+
+		return new IntermediateMappingsService(intermediaryTiny, expectedSrcNs);
```

Comment:
This doesnt seem right? Wouldnt you always expect IntermediateMappingsService to have the intermediary names as the source NS?

Reply:
The IntermediaryMappingsProvider extracts the mappings out of the jar as is, so the IntermediateMappingsService has the mappings as `official -> intermediary`. A custom intermediary provider for <1.3 would have to provide them as `intermediary -> [officialClient, officialServer]`.